### ISSUE TITLE
Search icon not displayed (iTerm2 + nerd-fonts)

### ISSUE THREAD

**Issue by denis-tchaikovsky** [AstroNvim/AstroNvim#812]
For some reason, the search icon is not displayed in my terminal (iTerm2). 

<img width="1053" alt="Screen Shot 2022-07-31 at 15 31 44" src="https://user-images.githubusercontent.com/12110348/182031706-f51f3ce8-ac95-442b-9f64-7633a77ef0f6.png">
<img width="1168" alt="Screen Shot 2022-07-31 at 15 31 59" src="https://user-images.githubusercontent.com/12110348/182031708-4b812060-8537-4325-8783-3ab50c7ba31d.png">

I tried installing multiple different nerd fonts but the issue stays the same. I saw the issue https://github.com/AstroNvim/AstroNvim/issues/677 From my understanding the suggested solution is to use another terminal, but I would like to find a way to make it work in iTerm2.

Would greatly appreciate any suggestions on how to fix this.

Thanks!

**Comment by mehalter** [comment]
Could you try some other nerd fonts? This seems to be an improper nerd tree patched font, maybe patched with an old version of nerd don't and doesn't have the most up to date icons.

**Comment by rontogiannis** [comment]
Same issue here, have tried a number of different nerd fonts, none of them seems to have the search icon.

**Comment by denis-tchaikovsky** [comment]
Hey @mehalter, thanks for the suggestion.
 
I tried so far:
- Fira Mono Nerd
- Fira Code Nerd
- Space Mono Nerd
- Hack Nerd
- SauceCodePro Nerd
- and few others

But unfortunately, none of them can display search icon 😅 

Any specific Nerd font you think can work?

**Comment by mehalter** [comment]
Hm, where are you installing these from @denis-tchaikovsky ? I would like to take a look at what verion of Nerd Font they have been patched with. Maybe we can figure out what the requirement is. It's really annoying to have a dependency on a specific patch version of nerd fonts tho :/ I took a look at the other magnifying glass icons and the Font Awesome one definitely looks the best with our other icons :( but maybe it can't be helped.

**Comment by denis-tchaikovsky** [comment]
@mehalter good point 👍

I'm installing them using `brew`. For example with `brew install --cask font-hack-nerd-font` 

They all seem to have version `2.1.0`

<img width="778" alt="Screen Shot 2022-08-02 at 22 06 41" src="https://user-images.githubusercontent.com/12110348/182473378-827e6815-fbad-406d-9859-b0b29d49ad21.png">

(["From" url](https://github.com/Homebrew/homebrew-cask-fonts/blob/HEAD/Casks/font-hack-nerd-font.rb))

From my understanding `2.1.0` is the latest stable in [Nerd Fonts](https://github.com/ryanoasis/nerd-fonts), right?

Do you think using the older version can fix this? 

Totally, Font Awesome looks really nice.

**Comment by mehalter** [comment]
lol now that is quite curious @denis-tchaikovsky I just tried installing all of these version 2.1.0 and the search icon works for all of them... Here is with Hack Nerd Font v2.1.0:


![2022-08-02_17:28:31_screenshot](https://user-images.githubusercontent.com/1591837/182476700-5f976ed1-c17e-4a04-846c-c9b085daf1d1.png)

**Comment by azinsharaf** [comment]
i am experiencing same issue in iTerms.
maybe take a look at this: https://apple.stackexchange.com/questions/395940/question-marks-in-iterm-font-issue

hyper on left / iterms on right, same font
<img width="1701" alt="image" src="https://user-images.githubusercontent.com/10101468/182481948-5db41cd3-2111-43f4-aa7d-7b5379ff641b.png">

**Comment by mehalter** [comment]
@azinsharaf in your screenshot I can see that iterm on the right and hyper on the left are using 2 different fonts. Could you try using the font that Hyper is loading on iterm to see if that fixes the issue? I'm not sure what font iTerm is loading there on the right, but it definitely looks very different

**Comment by mehalter** [comment]
It looks like the link that you provided the answer to the stack overflow question was "set a font that supports the glyph in iTerm"

**Comment by azinsharaf** [comment]
@mehalter i am sure i am using the same font on both: "JetbrainsMono Nerd Font".

**Comment by mehalter** [comment]
That is definitely strange. Are you setting it inside a profile directly in iTerm and making sure that it is active? It seems like there is an upstream bug with iTerm2 then that is causing it to not load the proper glyphs when they are available in the font.

**Comment by azinsharaf** [comment]
this is my iterm setup.

![Screen Shot 2022-08-02 at 3 55 46 PM](https://user-images.githubusercontent.com/10101468/182487278-7a69bd75-c19b-4e78-b479-381e6f1b1a6a.png)

**Comment by mehalter** [comment]
That is definitely very strange... @denis-tchaikovsky are you able to view the glyph correctly in another terminal like hyper?

**Comment by azinsharaf** [comment]
@denis-tchaikovsky 
Just in case you need to set the fonts in hyper, you edit this part in `.hyper.js` file. 

```
// font family with optional fallbacks
fontFamily: '"JetBrainsMono Nerd Font",
```

**Comment by denis-tchaikovsky** [comment]
@mehalter, yes, I did switch to the new font inside a profile directly in iTerm2.

<img width="945" alt="Screen Shot 2022-08-03 at 09 56 27" src="https://user-images.githubusercontent.com/12110348/182568215-dad7f43a-af79-4b44-93d7-837ed8e2a188.png">

I tried toggling the settings options with checkboxes in different ways, but it doesn't change the situation. 

I just tried Hyper and icons indeed displayed well there 🤔.
@azinsharaf thanks for the settings example 👍 

<img width="1673" alt="Screen Shot 2022-08-03 at 10 18 24" src="https://user-images.githubusercontent.com/12110348/182572814-b5c951ff-483f-488e-a54e-a096d9366b16.png">

That's an interesting issue. Could be some settings or some 3rd party that iTerm2 relies on.
@mehalter as far as I understand icons displayed well on your iTerm2, right? 

My iTerm2 version `3.4.16`
<img width="266" alt="Screen Shot 2022-08-03 at 10 27 42" src="https://user-images.githubusercontent.com/12110348/182574609-9e04c111-163f-4233-a326-a49c0f88a889.png">

Do you use the latest too?

**Comment by mehalter** [comment]
@denis-tchaikovsky I don't actually use iTerm because I don't use a Mac. I do think that there are some iTerm users on the discord. Let me send out a ping there and ask. I use kitty terminal on Linux.

**Comment by denis-tchaikovsky** [comment]
@mehalter, that makes sense. Good idea 👍 
Thanks for trying to help and also thanks a lot for such an amazing project 👏

**Comment by mehalter** [comment]
I'm not sure what else I can do with this issue :( I'm going to go ahead and close this and mark it as incomplete. Hopefully someone in the community can find a fix or report back any new information. Sorry!

Also thank you for the kind words @denis-tchaikovsky :)

**Comment by denis-tchaikovsky** [comment]
Sounds good. Hopefully, someone will find the fix. And no worries, I understand. It's a tricky one. Thanks, @mehalter 👍

**Comment by phturb** [comment]
I've found a temporary solution/fix.

You can replace the icon for a similar icon to replace the prompt icon. I recommend using the `nf-oct-search` icon instead of the `nf-fa-search` icon. This one will work on iTerm2.
`lua/configs/telescope.lua`

```lua
telescope.setup(astronvim.user_plugin_opts("plugins.telescope", {
  defaults = {
    prompt_prefix = " ",
    selection_caret = "❯ ",
...
```

<img width="481" alt="image" src="https://user-images.githubusercontent.com/43208288/182749064-63b52b68-317b-40c2-8c14-0dee582e879a.png">

**Comment by denis-tchaikovsky** [comment]
@phturb That's awesome, thank you for the fix! 🎉

@mehalter, the suggested `nf-oct-search` icon works great in my iTerm2 as well! Do you think we can replace it? 

@phturb I left the comment with a suggestion in your [PR](https://github.com/AstroNvim/AstroNvim/pull/824).

**Comment by mehalter** [comment]
Yeah this was one of the icons I tested @denis-tchaikovsky and it looks alright I guess. Nowhere near as nice looking as the font awesome icon when paired with the other icons. But I suppose it can't be helped, better to have compatibility I suppose. Have any of the iTerm2 users submitted a bug report? Back in like 2014 was the last time I used iTerm2 and the developers were very quick to fix bug reports back then at least 😂

**Comment by AVMusorin** [comment]
I faced the same problem with ITerm 3.4.19.
My solution is:
1. download `JetBrainsMono Nerd Font` and `Symbols Nerd Font` from [nerdfonts.com](https://www.nerdfonts.com/font-downloads)
2. use `Font Book` app on Mac to add them
3. Change Font in ITerm preferences 
<img width="940" alt="image" src="https://user-images.githubusercontent.com/17856418/216935307-6397816f-bdf4-4618-a911-e8647b2d91ef.png">

**Comment by JezerM** [comment]
This seems to be an iTerm2 issue, as stated here: https://gitlab.com/gnachman/iterm2/-/issues/10133. The solution would be to use the latest beta release (which has solved this issue) or wait for iTerm2 v3.5.0.
